OpenRA.Mods.Common/Traits/PaletteEffects/GlobalLightingPaletteEffect.cs

```diff
@@ -0,0 +1,82 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Drawing;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Used for day/night effects.")]
+	class GlobalLightingPaletteEffectInfo : ITraitInfo
+	{
+		[Desc("Do not modify graphics that use any palette in this list.")]
+		public readonly string[] ExcludePalettes = { "cursor", "chrome", "colorpicker", "fog", "shroud", "alpha" };
+
+		[Desc("Only modify graphics that use any of these palettes. If this list has members `ExcludePalettes` is ignored.")]
+		public readonly string[] OnlyPalettes = { };
+
+		[Desc("Do not modify graphics that start with these letters.")]
+		public readonly string[] ExcludePalettePrefixes = { };
+
+		[Desc("Only modify graphics that start with these letters.")]
+		public readonly string[] OnlyPalettePrefixes = { };
```

Comment:
This was initially required to do https://github.com/OpenRA/OpenRA/pull/7876 which I first talked him to do in it's own trait and then you came and found some fundamental problems. Will reduce the configurability here again.